### PR TITLE
Bind carousel example

### PR DESCRIPTION
**Pull request by kmh287**
Add amp-bind examples to Image Galleries With Carousel Example

### CODE REVIEW

**Comment by kmh287** [comment]
issue #579

File: src/30_Advanced/Image_Galleries_with_amp-carousel.html [hunk 1]
```diff
@@ -125,6 +135,53 @@
       </div>
     </div>
   </amp-carousel>
+  <!-- #### Linked Carousels with Bind -->
```
Comment (choumx) [inline src/30_Advanced/Image_Galleries_with_amp-carousel.html]: s/Bind/amp-bind

File: src/30_Advanced/Image_Galleries_with_amp-carousel.html [hunk 2]
```diff
@@ -125,6 +135,53 @@
...
+  <!--
+  Whenever the carousel slide changes, it updates `selectedSlide`. This causes the `text`
+  attribute in the `<p>` tag above the carousel to update as well.
+  -->
+  <div>
+    <p [text]="'Selected slide: ' + selectedSlide + '/2'">Selected slide: None<p>
+
+    <amp-carousel controls type=slides width=400 height=300
+    [slide]="selectedSlide" on="goToSlide:AMP.setState(selectedSlide=event.index)">
```
Comment (choumx) [inline src/30_Advanced/Image_Galleries_with_amp-carousel.html]: s/goToSlide/slideChange

https://github.com/ampproject/amphtml/pull/7387

File: src/30_Advanced/Image_Galleries_with_amp-carousel.html [hunk 3]
```diff
@@ -125,6 +135,53 @@
       </div>
     </div>
   </amp-carousel>
+  <!-- #### Linked Carousels with Bind -->
+  <!--
+  Whenever the carousel slide changes, it updates `selectedSlide`. This causes the `text`
+  attribute in the `<p>` tag above the carousel to update as well.
+  -->
+  <div>
+    <p [text]="'Selected slide: ' + selectedSlide + '/2'">Selected slide: None<p>
```
Comment (choumx) [inline src/30_Advanced/Image_Galleries_with_amp-carousel.html]: Probably should make this 1-indexed instead of 0-indexed.

Comment (sebastianbenz) [inline src/30_Advanced/Image_Galleries_with_amp-carousel.html]: the `p` closing tag is invalid

File: src/30_Advanced/Image_Galleries_with_amp-carousel.html (hunk 1 above)
Comment (kmh287) [inline src/30_Advanced/Image_Galleries_with_amp-carousel.html]: Done.

File: src/30_Advanced/Image_Galleries_with_amp-carousel.html [hunk 4]
```diff
@@ -125,6 +135,53 @@
       </div>
     </div>
   </amp-carousel>
+  <!-- #### Linked Carousels with Bind -->
+  <!--
```
Comment (kmh287) [inline src/30_Advanced/Image_Galleries_with_amp-carousel.html]: Done.

File: src/30_Advanced/Image_Galleries_with_amp-carousel.html (hunk 3 above)
Comment (kmh287) [inline src/30_Advanced/Image_Galleries_with_amp-carousel.html]: Done. Thanks for catching that.

File: src/30_Advanced/Image_Galleries_with_amp-carousel.html (hunk 2 above)
Comment (kmh287) [inline src/30_Advanced/Image_Galleries_with_amp-carousel.html]: Done.

File: src/30_Advanced/Image_Galleries_with_amp-carousel.html [hunk 5]
```diff
@@ -125,6 +135,53 @@
...
+
+    <amp-carousel controls type=slides width=400 height=300
+    [slide]="selectedSlide" on="goToSlide:AMP.setState(selectedSlide=event.index)">
+      <amp-img src="https://ampbyexample.com/img/image1.jpg" width=400 height=300></amp-img>
+      <amp-img src="https://ampbyexample.com/img/image2.jpg" width=400 height=300></amp-img>
+      <amp-img src="https://ampbyexample.com/img/image3.jpg" width=400 height=300></amp-img>
+    </amp-carousel>
+  </div>
+  <!--
+  These two carousels are linked to each other and to the large carousel above. Notice how
```
Comment (kmh287) [inline src/30_Advanced/Image_Galleries_with_amp-carousel.html]: Done.

File: src/30_Advanced/Image_Galleries_with_amp-carousel.html [hunk 6]
```diff
@@ -125,6 +135,53 @@
...
+  selecting an image on either of these two carousels changes the image on the top carousel.
+
+  In the first example, the CSS class for the selected image on the carousel is
+  set by bind when `selectedSlide` changes.
+  -->
+  <div>
```
Comment (kmh287) [inline src/30_Advanced/Image_Galleries_with_amp-carousel.html]: Done.

File: src/30_Advanced/Image_Galleries_with_amp-carousel.html [hunk 7]
```diff
@@ -125,6 +135,50 @@
...
+  <!-- #### Linking Carousels with amp-bind -->
+  <!--
+
+  With [amp-bind](https://github.com/ampproject/amphtml/blob/master/extensions/amp-bind/amp-bind.md) it's possible to link carousel state across multiple amp-carousels.
+
+  Whenever the carousel slide changes, it updates `selectedSlide`. This causes the `text`
+  attribute in the `<p>` tag above the carousel to update as well.
+  -->
+  <div>
+    <p> Selected slide: <span [text]="(selectedSlide + 1) + '/3'">1/3</span></p>
```
Comment (sebastianbenz) [inline src/30_Advanced/Image_Galleries_with_amp-carousel.html]: Can you move the `/3` outside the span as well?

**Comment by kul3r4** [comment]
Which version of AMP are you using?
There was a known [bug](https://github.com/ampproject/amphtml/pull/7229) where slide 0 (1 in human word) was never selected and it was fixed, but with version 1486153081489 it doesn't work, so this sample will never select slide 0 after selecting slide 1 and 2.
@sebastianbenz could you verify if it works for you?

**Comment by kul3r4** [comment]
I confirm that 1486153081489 doesn't contain the fix. I guess this fix is in canary (1486166232176) but canary at the moment gives another error: 
chunk.js:351 Uncaught (in promise) TypeError: Cannot read property 'enqueue' of undefined
    at lb (chunk.js:351)
    at jb (chunk.js:331)
    at chunk.js:72
    at Tb (bind-impl.js:196)
    at Rb (bind-impl.js:138)
    at bind-impl.js:101
so basically we shouldn't merge this code until either the fix is in master or canary is fixed.

**Comment by kul3r4** [comment]
@ithinkihaveacat FYI

**Comment by choumx** [comment]
The fixes for both bugs will be in this week's canary (should be deployed sometime later today). Note that after deployment, master will still contain the "cannot read property" bug for another week. Sorry about that!

**Comment by kul3r4** [comment]
Sounds good, thanks! If it works with dev channel we can always deploy it and just add a temporary note on the sample. My concern is that, being canary, it could break again but given it's just a week of waiting, that would be fine for me. I will test again tomorrow and update this thread.

File: src/30_Advanced/Image_Galleries_with_amp-carousel.html (hunk 7 above)
Comment (kmh287) [inline src/30_Advanced/Image_Galleries_with_amp-carousel.html]: Do you mean take the "/3" out of the span and replace the "1/3" in the p tag with "/3"?

Comment (sebastianbenz) [inline src/30_Advanced/Image_Galleries_with_amp-carousel.html]: Exactly (to remove as much duplication between binding and initial state as possible):

```
<p> Selected slide: <span [text]="(selectedSlide + 1)">1</span>/3</p>
```

Comment (kmh287) [inline src/30_Advanced/Image_Galleries_with_amp-carousel.html]: Done.

**Comment by sebastianbenz** [comment]
Thanks a lot!

**Pull request merged**